syntax = "proto3";
package backuprs;

// Registry server messages
service RegistryServer {
  // Registers a replica as well as sends the details of primary replica
  // Registers as primary replica if none exists
  rpc RegisterReplica (ReplicaDetails) returns (ReplicaDetails) {}
  // Returns List of replica's and their addresses
  rpc GetReplicaList (ClientDetails) returns (ReplicaList) {}
  // Returns the primary replica details 
  rpc GetPrimaryReplica (Empty) returns (ReplicaDetails) {}
}

// Define empty message
message Empty {}

message ClientDetails {
  string client_uuid = 1;
}

// Server details containing name and address
message ReplicaDetails {
  string name = 1;
  string addr = 2;
}

// List of replicas
message ReplicaList {
  // array of server details
  repeated ReplicaDetails replica_list = 1;
}